lib/logstash/inputs/kinesis/worker.rb

```diff
@@ -45,12 +65,25 @@ def checkpoint(checkpointer)
   end
 
   def process_record(record)
-    raw = @decoder.decode(record.getData).to_s
-    @codec.decode(raw) do |event|
-      @decorator.call(event)
-      @output_queue << event
+    @codec.decode(record) do |event|
```

Comment:
This is the line i'm referring to, to be clear. If `cloudwatch_logs` is set and `codec` is set to anything but `json`, this won't work properly.